src/sonic-yang-models/yang-models/sonic-nat.yang

```diff
@@ -0,0 +1,284 @@
+module sonic-nat {
+    namespace "http://github.com/Azure/sonic-nat";
+    prefix snat;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-acl {
+        prefix sacl;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC NAT yang model";
+
+    revision 2021-03-14 {
+        description
+            "Initial revision.";
+    }
+
+    typedef nat-clear-param {
+        type enumeration {
+            enum ENTRIES {
+                description "NAT dynamic Entries";
+            }
+            enum STATISTICS {
+                description "NAT Statistics";
+            }
+        }
+    }
+
+    typedef nat-type {
+        type enumeration {
+            enum snat {
+                description "Source NAT.";
+            }
+            enum dnat {
+                description "Destination NAT.";
+            }
+        }
+    }
+
+    typedef ip-protocol-type {
+        type enumeration {
+            enum TCP;
+            enum UDP;
+        }
+    }
+
+    typedef mode {
+        type enumeration {
+            enum enabled;
+            enum disabled;
+        }
+    }
+
+    typedef ip-address {
+        type union {
+          type inet:ipv4-address;
+          type string {
+                          pattern '(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])(-(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9]))?';
+                      }
+        }
+
+        description
+          "This type denotes a single IP address or a range of addresses.";
+    }
+
+    container sonic-nat {
+
+        container STATIC_NAPT {
+
+            list STATIC_NAPT_LIST {
```

Comment:
Max element in a yang list was part of RFC 6020, so I think we do not need to specify any version for this. But as I know there was a bug in Libyang library related to Max elements. 
Make sure that, you are not hitting the same bug.

Reply:
Yes, added max-elements to NAT_POOL, NAT_BINDINGS and NAT_GLOBAL tables.